src/main.rs

```diff
@@ -24,6 +24,20 @@ use {
     esp_idf_sys::esp_wifi_set_ps,
 };
 
+macro_rules! generate_register_modules {
+    ($($module:ident),*) => {
+        fn register_modules(registry: &mut ComponentRegistry) -> anyhow::Result<(), RegistryError> {
```

Comment:
This may need to be `_registry` in case there aren't actually any modules in play or you will get a warning. Or maybe a suppression?

Reply:
add 
`#[allow(unused_variables)]`

Reply:
Thanks; done.